src/service/variable-source.js

```diff
@@ -30,7 +31,42 @@ export let AsyncResolverDef;
 /** @typedef {{sync: SyncResolverDef, async: AsyncResolverDef}} */
 let ReplacementDef;
 
-const LOAD_EVENT_END = 'loadEventEnd';
+/**
+ * A list of events that the navTiming needs to wait for.
+ * Sort event in order
+ * @enum {number}
+ */
+const WAITFOR_EVENTS = {
+  VIEWER_FIRST_VISIBLE: 1,
+  DOCUMENT_COMPLETE: 2,
+  LOAD: 3,
+  LOAD_END: 4,
+};
+
+const NAV_TIMING_WAITFOR_EVENTS = {
+  // ready on viewer first visible
+  'navigationStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'redirectStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'redirectEnd': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'fetchStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'domainLookupStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'domainLookupEnd': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'connectStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'secureConnectionStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'connectEnd': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'requestStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'responseStart': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
+  'responseEnd': WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE,
```

Comment:
This size increase is coming from this struct. Can we make `VIEWER_FIRST_VISIBLE` the default, so that it doesn't need to be specified?

Reply:
True. I thought about that as well because there are many events that are waiting for `VIEWER_FIRST_VISIBLE`. 
However I don't know if there will be other performanceTiming events added in the future, or ones that we fail to include here. Because of that I wanted to make the default `WAITFOR_EVENTS` to be the `WAITFOR_EVENTS.LOAD` instead of `WAITFOR_EVENTS.VIEWER_FIRST_VISIBLE` as that's the safest fallback. 

I'm still open to this size optimization if you think that our current list of `performanceTiming` signals are complete, and there won't be frequent change to it in the future. Please let me know thanks.